### PR TITLE
Cleanup storehaus-core

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

File: storehaus-core/src/main/scala/com/twitter/storehaus/CollectionOps.scala [hunk 1]
```diff
@@ -28,11 +28,12 @@ object CollectionOps {
    * PivotEncoder to Map[InnerK, V] (vs Iterable[(InnerK, V)], as
    * com.twitter.bijection.Pivot would provide).
    */
-  def pivotMap[K, OuterK, InnerK, V](pairs: Map[K, V])(split: K => (OuterK, InnerK)): Map[OuterK, Map[InnerK, V]] =
+  def pivotMap[K, OuterK, InnerK, V](pairs: Map[K, V])(
+      split: K => (OuterK, InnerK)): Map[OuterK, Map[InnerK, V]] =
     pairs.toList.map {
```
Comment (johnynek) [inline storehaus-core/src/main/scala/com/twitter/storehaus/CollectionOps.scala]: while we are at it can we do: `pairs.iterator.map { }.toList.groupBy` which might be marginally faster.

File: storehaus-core/src/main/scala/com/twitter/storehaus/FutureCollector.scala [hunk 2]
```diff
@@ -28,12 +28,12 @@ object FutureCollector {
   /**
    * If any future fails, the remaining future fails.
    */
-  implicit def default = new FutureCollector {
+  implicit def default: FutureCollector = new FutureCollector {
     override def apply[T](futureSeq: Seq[Future[T]]) = Future.collect(futureSeq)
```
Comment (johnynek) [inline storehaus-core/src/main/scala/com/twitter/storehaus/FutureCollector.scala]: no return type here?

File: storehaus-core/src/main/scala/com/twitter/storehaus/IterableStore.scala [hunk 3]
```diff
@@ -1,36 +1,33 @@
...
+import com.twitter.util.Future
 import com.twitter.concurrent.Spool
-import com.twitter.concurrent.Spool.*::
 
 object IterableStore {
   /** Factory method to create a IterableStore from a Map. */
   def fromMap[K, V](m: Map[K, V]): IterableStore[K, V] = new MapStore(m)
 
   /** Helper method to convert Iterator to Spool. */
   def iteratorToSpool[K, V](it: Iterator[(K, V)]): Future[Spool[(K, V)]] = Future.value {
```
Comment (johnynek) [inline storehaus-core/src/main/scala/com/twitter/storehaus/IterableStore.scala]: while we are here, I think it is binary compatible (though not source) to change to:

``` scala
def iteratorToSpool[T](it: Iterator[T]): Future[Spool[T]] = ...
```

Not sure why this forces `(K, V)`

File: storehaus-core/src/main/scala/com/twitter/storehaus/MapStore.scala [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
 
 package com.twitter.storehaus
 
+import com.twitter.concurrent.Spool
```
Comment (johnynek) [inline storehaus-core/src/main/scala/com/twitter/storehaus/MapStore.scala:19]: why is this needed? I don't see it used by name?

File: storehaus-core/src/main/scala/com/twitter/storehaus/MapStore.scala [hunk 5]
```diff
@@ -25,7 +26,7 @@ import com.twitter.util.Future
  */
 class MapStore[K, +V](val backingStore: Map[K, V] = Map[K, V]()) extends ReadableStore[K, V]
     with IterableStore[K, V] {
-  override def get(k: K) = Future.value(backingStore.get(k))
+  override def get(k: K): Future[Option[V]] = Future.value(backingStore.get(k))
 
-  override def getAll = IterableStore.iteratorToSpool(backingStore.iterator)
+  override def getAll: Future[Spool[(K, V)]] = IterableStore.iteratorToSpool(backingStore.iterator)
```
Comment (BenFradet) [inline storehaus-core/src/main/scala/com/twitter/storehaus/MapStore.scala:31]: it's used here

Comment (johnynek) [inline storehaus-core/src/main/scala/com/twitter/storehaus/MapStore.scala:31]: Ahh.

**Comment by johnynek** [comment]
Thank you sir!
